File: Quick-Sort.c
#include<stdio.h>

int Partion(int a[] , int l , int r ){
  int pivot=a[l];
  int i=l,j=r,t;
  while(i<=j){
    while(a[i]<=pivot &&i<=r)
    {
      i++;
    }
    while(a[j]>pivot)
    {
      j--;
    }
    if(i<j)
    {
      t=a[j];
      a[j]=a[i];
      a[i]=t;
    }
    t=a[j];
    a[j]=a[l];
    a[l]=t;
    return j;
  }
}

int QuickSort(int a[], int L , int R){
  int q ;
  if(L < R){
    q = Partion(a , L , R );
    QuickSort(a , L , q-1 );
    QuickSort(a , q+1 , R );
  }
}

void main(){
  int a[25] , n , i , Left , Right ;
  printf("Enter the no of elements: ");
  scanf("%d",&n);
  for(i = 0 ; i < n ; i++){
    printf("Enter the Element: ");
    scanf("%d",&a[i]);
  }
  Left = 0 ;
  Right = n-1 ;

  QuickSort(a, Left, Right);
  printf("The Sort array is ");
  for(i = 0 ; i < n ; i++ ){
    printf("%d ",a[i]);
  }
}

File: QuickSort.java
import java.util.*;
import java.io.*;
class QuickSort
{
ArrayList&lt;String&gt; names = new ArrayList&lt;String&gt;();
void inserttoList()
{
Scanner in = new Scanner(System.in);
System.out.println(&quot;Storing Names\n&quot;);
do
{
System.out.println(&quot;Enter name&quot;);
names.add(in.nextLine());
System.out.println(&quot;Continue entry ? (y/n)&quot;);
}
while(Character.toLowerCase(in.nextLine().charAt(0))==&#39;y&#39;);
}
void printList()
{
System.out.println(&quot;Printing Stored Names&quot;);
for (String s:names)
{
System.out.print(s+&quot;\t&quot;);
}
System.out.println();
}
void sort(int left,int right)
{
String pivot;
int i,j;
if (left&gt;=right)
return;
pivot = names.get(right);
i = left;
j = right;
while (i&lt;j)
{
while(names.get(i).compareTo(pivot) &lt; 0)
i++;
while(names.get(j).compareTo(pivot) &gt; 0)
j--;
if (i&lt;j);
{
Collections.swap(names, i, j);
}
}
sort(left, j-1);
sort(i+1, right);
}

void startSort()
{
sort(0,(names.size()-1));
System.out.println(&quot;\nSORTING\n&quot;);
}
public static void main(String[] args)
{
Scanner in = new Scanner(System.in);
QuickSort ob = new QuickSort();
ob.inserttoList();
ob.printList();
ob.startSort();
ob.printList();
}
}
